mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/mail/InMemoryMailboxMapper.java

```diff
@@ -134,12 +135,15 @@ public Flux<Mailbox> list() {
 
     @Override
     public Mono<ACLDiff> updateACL(Mailbox mailbox, MailboxACL.ACLCommand mailboxACLCommand) {
-        return Mono.fromCallable(() -> {
-            MailboxACL oldACL = mailbox.getACL();
-            MailboxACL newACL = mailbox.getACL().apply(mailboxACLCommand);
-            mailboxesByPath.get(mailbox.generateAssociatedPath()).setACL(newACL);
-            return ACLDiff.computeDiff(oldACL, newACL);
-        });
+        return Mono.just(mailbox.getACL())
+            .flatMap(oldACL -> Mono.fromCallable(() -> mailboxesByPath.compute(mailbox.generateAssociatedPath(), (path, existingMailbox) -> {
```

Comment:
The fact that mailboxes are indexed by path, which is mutable, creates unecessary complexity.

I'm then also 100% sure that there's a data race between ACL update and mailbox renames that WILL bite back when implementing right propagation upon renames of team mailboxes.

I would seriously advise using `mailboxesById` which would be a Map<MailboxId, Mailbox> instead...